### ISSUE TITLE
[Bug]: missing trace for 2nd browser 

### ISSUE THREAD

**Issue by dimkin-eu** [microsoft/playwright#31601]
### Version

1.45.1

### Steps to reproduce

```js
test.describe("describe", () => {
    test("short spec", async ({page,browser}) => {
        await page.goto('https://playwright.dev/')
        const page2 = await (await browser.newContext()).newPage()
        await page2.goto('https://github.com/microsoft/playwright')
        await page.locator("bazinga").click()
        await page2.context().close()
    })
})
```

### Expected behavior

- test fails
- trace contains all browsers from test

![Screenshot 2024-07-09 at 09 49 19](https://github.com/microsoft/playwright/assets/7899764/ec300388-0725-457e-8bdd-5d7a597b35a9)

### Actual behavior

- test fails ( this is fine )
- (it doesn't matter in which browser test fails) trace contains only trace from 'page' fixture

![Screenshot 2024-07-09 at 09 49 30](https://github.com/microsoft/playwright/assets/7899764/2e4160a6-fc58-40e2-9023-1f9ad9476d67)

### Additional context

1.44.1 was fine, 1.45.1 - fails

    use: {
        // headless: false,
        ignoreHTTPSErrors: true,
        screenshot: "off",
        video: "off",
        trace: "on",
        contextOptions: {
            bypassCSP: true,
        },
    },

### Environment

```shell
System:
    OS: macOS 14.5
    CPU: (12) x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
    Memory: 20.10 MB / 32.00 GB
  Binaries:
    Node: 20.13.1 - /usr/local/bin/node
    npm: 10.5.2 - /usr/local/bin/npm
    pnpm: 9.4.0 - /usr/local/bin/pnpm
  Languages:
    Bash: 3.2.57 - /bin/bash
  npmPackages:
    @playwright/test: 1.45.1 => 1.45.1
```

**Comment by dimkin-eu** [comment]
for sure related to https://github.com/microsoft/playwright/issues/31541

**Comment by mxschmitt** [comment]
Looks like already fixed on ToT - @dgozman should we cherry-pick?

**Comment by dimkin-eu** [comment]
hmm, there were case which sounds exactly like this , but looks like that tests didn't found ( one more ) regression :(
https://github.com/microsoft/playwright/issues/23421

**Comment by mxschmitt** [comment]
I can verify this is fixed on @beta and should be fixed in v1.45.2 shortly.
